packages/kbn-dev-utils/src/code_ownership/ownership_config.ts

```diff
@@ -0,0 +1,658 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export interface OwnershipRule {
+  files: string[]; // Files listed in CODEOWNERS and Code Coverage
+  excludeFiles: string[]; // Files in CODEOWNERS, with a ! prepended
+  codeOwner?: string; // CODEOWNERS Team Name
+  coverageOwner?: string; // Code Coverage Team Name
+}
+
+export const rules: OwnershipRule[] = [
+  {
+    files: [
+      '/x-pack/plugins/dashboard_enhanced/',
+      '/x-pack/plugins/discover_enhanced/',
+      '/x-pack/plugins/lens/',
+      '/x-pack/plugins/graph/',
+      '/src/plugins/dashboard/',
+      '/src/plugins/discover/',
+      '/src/plugins/input_control_vis/',
+      '/src/plugins/kibana_legacy/',
+      '/src/plugins/vis_default_editor/',
+      '/src/plugins/vis_type_markdown/',
+      '/src/plugins/vis_type_metric/',
+      '/src/plugins/vis_type_table/',
+      '/src/plugins/vis_type_tagcloud/',
+      '/src/plugins/vis_type_timelion/',
+      '/src/plugins/vis_type_timeseries/',
+      '/src/plugins/vis_type_vega/',
+      '/src/plugins/vis_type_vislib/',
+      '/src/plugins/vis_type_xy/',
+      '/src/plugins/visualize/',
+    ],
+    excludeFiles: [],
+    codeOwner: '@elastic/kibana-app',
+    coverageOwner: 'kibana-app',
```

Comment:
I this we should focus this PR on codeowner creation and I don't think we should worry about coverage ownership for now. 

I think it will be valuable to see all the coverage related changes in a single PR rather than having some of them here and some of them in a separate PR.